### PR TITLE
Make grant form changes requested in #248

### LINKED ISSUE #248: As a proposal creator, I need to understand the questions I’m being asked and know how to enter the information I need to provide so that I can submit a new proposal correctly and completely on the first try.
**Background**
This story covers a bunch of small tweaks to the proposal creation form that we learned about in Sprint Pine Cone Cap user research. This may need to be broken up into smaller issues, but here are the recommendations from the research

 **Project overview**
 - Include cooperator name in project overview
 - Include total amount of funds being requested in project overview
 - Provide link to help text to explain proposed start date fields (may need to break out into new issue)
 -  Remove “will this be covered by FFA?” 
 - Remove “advance allowed” 
 
**When adding partnership details**
 - Remove “new cooperator” question
 ~- Add explanatory text or in context glossary link to help folks understand DUNS#~ (new issue, #323)
 
~**When creating commitments**~
 (moved to new issue #325) 

**We'll know we're done when...**
- [ ] We have implemented refinements in our testing environment

**Tasks**
- [ ] Dev makes updates
- [ ] PO approved

### PR DESCRIPTION
**Pull request by beechnut**
- Adds and removes fields
- Adds some tests for these changes
- Makes the fields on the minimal grant form required and alters the help text for form creation context (removes a comparison to a field not on this form)

Closes #248

### CODE REVIEW

File: nrm_django/grants/forms.py [hunk 1]
```diff
@@ -15,17 +15,28 @@ class MinGrantForm(forms.ModelForm):
     Defines the minimum viable Grant/proposal creation form.
     """
 
+    # Require all fields in minimal grant creation form.
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        for key in self.Meta.fields:
+            self.fields[key].required = True
+        self.fields[
```
Comment (neilmb) [inline nrm_django/grants/forms.py:23]: How is this related to the field definition on grants/models.py L98?

File: nrm_django/grants/models.py [hunk 2]
```diff
@@ -361,6 +366,7 @@ class Grant(models.Model):
     )
     reroute_from = models.CharField(max_length=10, blank=True, null=True)
     reroute_date = models.DateField(blank=True, null=True)
+    # TODO: Is this typo in the source data?
```
Comment (neilmb) [inline nrm_django/grants/models.py]: Yes, the field name in the source database is this. We could improve on it by calling the field `certification_date` and using `db_column="certificaion_date"`.

File: nrm_django/grants/templates/includes/min_form.html [hunk 3]
```diff
@@ -1,40 +1,12 @@
...
-        <div id="rwu" class="usa-accordion__content usa-prose">
-            {% include "includes/field.html" with field=form.proj_rwu %}
-            {% include "includes/field.html" with field=form.journal_ind %}
-            {% include "includes/field.html" with field=form.proj_science_cd %}
-            {% include "includes/field.html" with field=form.research_type %}
-        </div>
-    </fieldset>
-</div>
-
+<br />
```
Comment (neilmb) [inline nrm_django/grants/templates/includes/min_form.html:11]: Are we using these two line breaks on purpose for spacing?

File: nrm_django/grants/tests/test_views.py [hunk 4]
```diff
@@ -64,3 +69,42 @@ def test_logged_in_uses_correct_template(self):
                 "grants/update.html",
                 f"The template grants/update.html was not used for grant {grant.cn}.",
             )
+
+    # These next 5 tests could be refactored into one containing all of the assertions.
```
Comment (neilmb) [inline nrm_django/grants/tests/test_views.py]: One way to DRY these out is to sub-class GrantUpdateViewTest, extend the `setUp` method with these five repeated lines and then just write the tests as assertions about `self.response`.

**Review by neilmb — DISMISSED** [review]
Mostly clarifications and DRY, but overall great to see how we can incorporate research-based changes.

File: nrm_django/grants/templates/includes/min_form.html (hunk 3 above)
Comment (beechnut) [reply nrm_django/grants/templates/includes/min_form.html:11]: Yes—removing the field that _was_ at the end also removed the spacing, so this was a quick-fix replacement.

File: nrm_django/grants/models.py (hunk 2 above)
Comment (beechnut) [reply nrm_django/grants/models.py]: Thanks. I didn't intend for this todo to make it in, but since we're here I may update that field name.

File: nrm_django/grants/forms.py (hunk 1 above)
Comment (beechnut) [reply nrm_django/grants/forms.py:23]: L23 onward removes the text "This date cannot be after the Expiration date." from this minimal grant creation form. I removed it because there is no expiration date field on this form, so the reference to it is confusing.

File: nrm_django/grants/tests/test_views.py (hunk 4 above)
Comment (beechnut) [reply nrm_django/grants/tests/test_views.py]: Thanks, I'll do that and update this PR.

File: nrm_django/grants/tests/test_views.py [hunk 5]
```diff
@@ -70,41 +70,26 @@ def test_logged_in_uses_correct_template(self):
...
-    # These next 5 tests could be refactored into one containing all of the assertions.
-    def test_proposal_creation_form_includes_cooperator(self):
-        request = self.factory.get(reverse("add_grant", kwargs={"app_label": "grants"}))
-        request.user = self.user
-        response = GrantCreateView.as_view()(request)
-        response.render()
-        self.assertIn(b"Applicant/Cooperator Name", response.content)
 
-    def test_proposal_creation_form_includes_total_funds(self):
+class GrantProposalCreationTest(GrantUpdateViewTest):
```
Comment (neilmb) [inline nrm_django/grants/tests/test_views.py:74]: These look much slimmer, thanks.

**Review by neilmb — APPROVED** [review]
